crates/rust-analyzer/src/task_pool.rs

```diff
@@ -45,11 +45,10 @@ impl<T> TaskPool<T> {
     }
 }
 
-/// `TaskQueue`, like its name suggests, queues tasks.
-///
-/// This should only be used if a task must run after [`GlobalState::process_changes`]
-/// has been called.
-pub(crate) struct TaskQueue {
-    pub(crate) sender: crossbeam_channel::Sender<QueuedTask>,
-    pub(crate) receiver: crossbeam_channel::Receiver<QueuedTask>,
+/// `DeferredTaskQueue` holds deferred tasks. These are tasks that
```

Comment:
one small nit: i'd keep the first line of the doc comment on a separate line that summarizes what this is.

after a new, empty line, add a paragraph describing what's happening.